test/Base_Tests/src/Semantic/Warnings_Spec.enso

```diff
@@ -277,11 +277,16 @@ add_specs suite_builder = suite_builder.group "Dataflow Warnings" group_builder-
 
         res . should_equal [1, 0, 1, 0]
         Warning.get_all res wrap_errors=True . map .value . should_equal [(Map_Error.Error 3 'warn4'), (Map_Error.Error 2 'warn3'), (Map_Error.Error 1 'warn2'), (Map_Error.Error 0 'warn1')]
-        res.at 0 . should_equal 1
-        Warning.get_all (res.at 0) . map .value . should_equal ['warn4', 'warn3', 'warn2', 'warn1']
+        elem0 = res.at 0
+        elem0 . should_equal 1
+        four_raw = Warning.get_all elem0
+        four_map = four_raw . map .value
+        four_map . should_equal ['warn4', 'warn3', 'warn2', 'warn1']
 
         slice = res.slice 1 4
-        Warning.get_all slice wrap_errors=True . map .value . should_equal [(Map_Error.Error 3 'warn4'), (Map_Error.Error 2 'warn3'), (Map_Error.Error 1 'warn2'), (Map_Error.Error 0 'warn1')]
+        slice_warn = Warning.get_all slice wrap_errors=True
```

Comment:
We need 91d3584 for this `slice` checks to pass:
- `slice` itself `hasWarnings`, but only internally
- `slice` isn't wrapped with `WithWarnings`
- the checks here rely on `slice` having the warnings only internally

Reply:
I tried to lookup the history of ArraySlice/Warnings but to no avail. I would really expect `hasWarnings` to work in this case. So I'm a bit surprised at that cast.

Reply:
OK, let's try to avoid direct use of `WithWarnings`:
- first of all we have to fix `EnsoHashMap`: 21d9bc3f63
- then `SliceArrayVectorNode` can `@AcceptsWarning`: e6af654e51
- and we can recognize `hasWarnings` in `BuiltinRootNode.ArgsNode`

Reply:
- e6af654 is causing benchmarks regression
- https://github.com/enso-org/enso/issues/12258#issuecomment-2652656902